src/libraries/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -262,4 +260,259 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
         public override int GetHashCode() =>
             GetType().GetHashCode();
     }
+
+    [TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
+    public static class EqualityComparer
+    {
+        // Equality comparison using sequence equality
+        public static IEqualityComparer<TEnumerable> CreateEnumerableComparer<TEnumerable, T>(IEqualityComparer<T>? elementComparer = null)
+            where TEnumerable : IEnumerable<T> =>
+            new EnumerableEqualityComparer<TEnumerable, T>(elementComparer);
+
+        public static IEqualityComparer<TSet> CreateSetComparer<TSet, T>(IEqualityComparer<T>? elementComparer = null)
+            where TSet : IReadOnlySet<T> =>
+            new SetEqualityComparer<TSet, T>(elementComparer);
+
+        public static IEqualityComparer<TDictionary> CreateDictionaryComparer<TDictionary, TKey, TValue>(IEqualityComparer<TKey>? keyComparer = null, IEqualityComparer<TValue>? valueComparer = null)
+            where TDictionary : IReadOnlyDictionary<TKey, TValue> =>
+            new DictionaryEqualityComparer<TDictionary, TKey, TValue>(keyComparer, valueComparer);
+    }
+
+    [Serializable]
+    [TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
+    // Needs to be public to support binary serialization compatibility
```

Comment:
This is a new type. There are no binary compatibility concerns with new types. We explicitly do not want to make any new types compatible with binary serialization.

The type should be internal and all `TypeForwardedFrom` and `Serializable` attributes should be deleted from this change.

Reply:
Ok, I was not sure.
Removed!